tilecloud_chain/__init__.py

```diff
@@ -838,11 +856,22 @@ def counter(self, size=False):
         self.imap(count)
         return count
 
-    def process(self, name=None):
-        if name is None:
-            name = self.layer.get('post_process')
-        if name is not None:
-            self.imap(Process(self.config['process'][name], self.options))
+    def process(self, name=None, key='post_process'):
+        if self.layer:
+            if name is None:
+                name = self.layer.get(key)
+            if name is not None:
+                self.imap(Process(self.config['process'][name], self.options))
+        else:
+            processes = {}
+            for lname, layer in self.layers.items():
+                name_ = name
+                if name_ is None:
+                    name_ = layer.get(key)
+                if name_ is not None:
+                    processes[lname] = Process(self.config['process'][name_], self.options)
+            if processes:
```

Comment:
If none of the layers have a process with a given key (e.g. if no layer has a post-processing step), processes will be falsy.